From f902c19da0ad5f35a3ce9733bc7e98412d0798de Mon Sep 17 00:00:00 2001
From: Antonio Sanchez <cantonios@google.com>
Date: Tue, 4 Feb 2025 10:25:43 -0800
Subject: [PATCH] Use dynamic version in pyproject.toml and remove unused
 header declaration from package data.

The version will be imported from `ml_dtypes.__version__`.

PiperOrigin-RevId: 723138894
---
 CHANGELOG.md          | 3 ++-
 ml_dtypes/__init__.py | 2 +-
 pyproject.toml        | 6 ++++--
 3 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/CHANGELOG.md b/CHANGELOG.md
index d7f8750d..e1b0519d 100644
--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -23,11 +23,12 @@ To release a new version (e.g. from `1.0.0` -> `2.0.0`):
 
 ## [Unreleased]
 
-## [0.5.2] - 2025-01-06
+## [0.5.2] - 2025-01-31
 
  * Dropped support for Power wheels again. These turned out to cause problems in
    our release process. We will consider readding these if NumPy ships Power
    wheels.
+ * Fixed GCC compilation issues related to ambiguous casts.
 
 ## [0.5.1] - 2025-01-06
 
diff --git a/ml_dtypes/__init__.py b/ml_dtypes/__init__.py
index 005a27a8..db38cd13 100644
--- a/ml_dtypes/__init__.py
+++ b/ml_dtypes/__init__.py
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-__version__ = "0.5.1"  # Keep in sync with pyproject.toml:version
+__version__ = "0.5.1"
 __all__ = [
     "__version__",
     "bfloat16",
diff --git a/pyproject.toml b/pyproject.toml
index c040eb32..efdc9e23 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,6 +1,6 @@
 [project]
 name = "ml_dtypes"
-version = "0.5.1"  # Keep in sync with ml_dtypes/__init__.py:__version__
+dynamic = ["version"]  # Load from ml_dtypes.__version__.
 description = ""
 readme = "README.md"
 requires-python = ">=3.9"
@@ -60,6 +60,8 @@ build-backend = "setuptools.build_meta"
 packages = ["ml_dtypes"]
 include-package-data = false
 
+[tool.setuptools.dynamic]
+version = {attr = "ml_dtypes.__version__"}
+
 [tool.setuptools.package-data]
 ml_dtypes = ["py.typed"]
-"ml_dtypes.include" = ["*.h"]
